shapely/geometry/point.py

```diff
@@ -64,12 +64,12 @@ def __new__(self, *args):
             # Accept either (x, y) or [(x, y)]
             if not hasattr(coords, "__getitem__"):  # generators
                 coords = list(coords)
-
             coords = np.asarray(coords).squeeze()
         else:
             # 2 or 3 args
             coords = np.array(args).squeeze()
 
+        coords = [float(c) for c in coords]
```

Comment:
We could avoid doing this if we already have a numeric numpy array. So maybe something like:

```
if not np.issubdtype(coords.dtype, np.number):
    coords = [float(c) for c in coords]
```

and related to the other comment, we could maybe also first check explicitly for >1d array case to provide a better error message (untested):

```
if coords.ndim > 1:
    raise ValueError("...")
```

Reply:
Done.